app/controllers/api/v1/rooms_controller.rb

```diff
@@ -51,13 +51,13 @@ def public_show
       def create
         # TODO: amir - ensure accessibility for authenticated requests only.
         # The created room will be the current user's unless a user_id param is provided with the request.
-        # user_id = room_params[:user_id]
-        room = Room.create(name: room_params[:name], user_id: room_params[:user_id])
+        @new_room = Room.create(name: room_params[:name], user_id: room_params[:user_id])
 
         return render_error errors: user.errors.to_a if hcaptcha_enabled? && !verify_hcaptcha(response: params[:token])
 
-        if room.save
-          logger.info "room(friendly_id):#{room.friendly_id} created for user(id):#{room.user_id}"
+        if @new_room.save
```

Comment:
The room should be already created,
I think a test like `persisted?` is more adequate.

Reply:
If it's feasible using `create!`  or changing the logic to first instantiate a room model `new` and then `save` it  may help